xdsl/backend/riscv/riscv_scf_to_asm.py

```diff
@@ -53,7 +53,7 @@ def match_and_rewrite(self, op: riscv_scf.ForOp, rewriter: PatternRewriter, /):
         # upper bound and branching to the loop body if the condition is met.
         rewriter.insert_op_before_matched_op(
             [
-                get_loop_var := riscv.GetRegisterOp(loop_var_reg),
+                get_loop_var := riscv.MVOp(op.lb, rd=loop_var_reg),
```

Comment:
When in a nested loop, the iteration index variable was never reset to the beginning of the loop.

Reply:
Should this not also update a filecheck test case?

Reply:
yep, fixing